sys/include/net/skald.h

```diff
@@ -0,0 +1,129 @@
+ * `broadcaster` role only. With this focus, the stack allows for setting up
+ * different kind of beacons using an extremely low memory footprint.
+ *
+ * # Design Decisions and Limitations
+ * - support for random public addresses only (generated using `luid` for now)
+ * - advertising interval is configured during compile time, override by setting
+ *   `CFLAGS+=-DSKALD_INTERVAL=xxx`
+ * - advertising channels are configured during compile time, override by
+ *   setting `CFLAGS+=-DSKALD_ADV_CHAN={37,39}`
+ *
+ * # Implementation state
+ * Supported:
+ * - advertising of custom GAP payloads
+ * - iBeacon (full support)
+ * - Eddystone (partly supported)
+ *
+ * Limitations:
+ * - currently Skald supports random static addresses only (generated using
+ *   the `luid` module)
+ *
+ * @{
+ * @file
+ * @brief       Skald's basic interface
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NET_SKALD_H
+#define NET_SKALD_H
+
+#include <stdint.h>
+
+#include "xtimer.h"
+#include "net/ble.h"
+#include "net/netdev/ble.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Static advertising interval
+ */
+#ifndef SKALD_INTERVAL
+#define SKALD_INTERVAL          (1 * US_PER_SEC)
+#endif
+
+/**
+ * @brief   Static list of used advertising channels
+ */
+#ifndef SKALD_ADV_CHAN
+#define SKALD_ADV_CHAN          { 37, 38, 39 }
+#endif
+
+/**
+ * @brief   UUID representation format used by Skald
+ */
+typedef struct {
+    uint8_t u8[16];         /**< UUID with byte-wise access */
+} skald_uuid_t;
+
+/**
+ * @brief   Advertising context holding the advertising data and state
+ */
+typedef struct {
+    netdev_ble_pkt_t pkt;   /**< packet holding the advertisement (GAP) data */
+    xtimer_t timer;         /**< timer for scheduling advertising events */
+    uint32_t last;          /**< last timer trigger (for offset compensation) */
+    uint8_t cur_chan;       /**< keep track of advertising channels */
+} skald_ctx_t;
+
+/**
+ * @brief   Initialize Skald and the underlying radio
+ *
+ * @param[in,out] dev   BLE capable radio device
+ */
+void skald_init(netdev_t *dev);
+
+/**
+ * @brief   Start advertising the given packet
+ *
+ * The packet will be send out each advertising interval (see SKALD_INTERVAL) on
+ * each of the defined advertising channels (see SKALD_ADV_CHAN).
+ *
+ * @param[in,out] ctx   start advertising this context
+ */
+void skald_adv_start(skald_ctx_t *ctx);
+
+/**
+ * @brief   Stop the ongoing advertisement
+ *
+ * @param[in,out] ctx   stop advertising this context
+ */
+void skald_adv_stop(skald_ctx_t *ctx);
+
+/**
+ * @brief   Generate a random public address
+ *
+ * @todo    Make sure resulting address does not conflict with and assigned
+ *          vendor prefix (however we will do that..)
+ *
+ * @note    @p buf must be able to hold BLE_ADDR_LEN (6) bytes
+ *
+ * @param[out] buf      the generated address is written to this buffer
+ */
+void skald_generate_random_addr(uint8_t *buf);
```

Comment:
`random` might not be a good term here, since the result of `get_luid()` is not random, but pretty predeterminable.

Reply:
but the bluetooth standard calls this type of address random.

Reply:
Ah, okay.